manticore/ethereum/manticore.py

```diff
@@ -572,6 +579,76 @@ def make_symbolic_arguments(self, types):
+            :type jfile: str or IOBase
+            :param owner: owner account (will be default caller in any transactions)
+            :type owner: int or EVMAccount
+            :param contract_name: Name of the contract to analyze (optional if there is a single one in the source code)
+            :type contract_name: str
+            :param balance: balance to be transferred on creation
+            :type balance: int or BitVecVariable
+            :param gas: gas budget for each contract creation needed (may be more than one if several related contracts defined in the solidity source)
+            :type gas: int
+            :param network_id: Truffle network id to instantiate
+            :rtype: EVMAccount
+        """
+
+        if isinstance(jfile, io.IOBase):
+            jfile = jfile.read()
+        elif not isinstance(jfile, str):
+            raise TypeError(f'source code bad type: {type(jfile).__name__}')
+
+        truffle = json.loads(jfile)
+        hashes = {}
+        for item in truffle['abi']:
+            item_type = item['type']
+            if item_type in ('function'):
+                signature = SolidityMetadata.function_signature_for_name_and_inputs(item['name'], item['inputs'])
+                hashes[signature] = sha3.keccak_256(signature.encode()).hexdigest()[:8]
+                if 'signature' in item:
+                    if item['signature'] != f'0x{hashes[signature]}':
+                        raise Exception(f"Something wrong with the sha3 of the method {signature} signature (a.k.a. the hash)")
+
+        if contract_name is None:
+            contract_name = truffle["contractName"]
+
+        if network_id is None:
+            if len(truffle['networks']) > 1:
+                raise Exception("Network id not specified")
+            if len(truffle['networks']) == 1:
+                network_id = list(truffle['networks'].keys())[0]
+        if network_id in truffle['networks']:
+            temp_dict = truffle['networks'][network_id]['links']
+            links = dict((k, int(v['address'], 0)) for k, v in temp_dict.items())
+        else:
+            links = ()
+
+        source_code = truffle["source"]
+        bytecode = self._link(truffle["bytecode"][2:], links)
+        runtime = self._link(truffle["deployedBytecode"][2:], links)
+        if "sourceMap" in truffle:
+            srcmap = truffle["sourceMap"].split(';')
+        else:
+            srcmap_runtime = []
+        if "deployedSourceMap" in truffle:
+            srcmap_runtime = truffle["deployedSourceMap"].split(';')
+        else:
+            srcmap_runtime = []
+        abi = truffle['abi']
+        md = SolidityMetadata(contract_name, source_code, bytecode, runtime, srcmap, srcmap_runtime, hashes, abi, b'')
+        contract_account = self.create_contract(owner=owner, balance=balance, init=md._init_bytecode)
+
+        if contract_account is None:
+            raise EthereumError("Failed to build contract %s" % contract_name_i)
```

Comment:
Typo?
```suggestion
            raise EthereumError("Failed to build contract %s" % contract_name)
```

Reply:
bad copy pasta Also we need a f-string there.